src/__tests__/integration/internal-writes/internal-write-aftr.test.ts

```diff
@@ -0,0 +1,287 @@
+/* eslint-disable */
+import fs from 'fs';
+
+import ArLocal from 'arlocal';
+import {JWKInterface} from 'arweave/node/lib/wallet';
+import path from 'path';
+import {mineBlock} from '../_helpers';
+import {Contract} from '../../../contract/Contract';
+import {Warp} from '../../../core/Warp';
+import {WarpFactory} from '../../../core/WarpFactory';
+import {LoggerFactory} from '../../../logging/LoggerFactory';
+
+interface ExampleContractState {
+  counter: number;
+  errorCounter: number;
+}
+
+describe('Testing internal writes', () => {
+  let aftrContractSrc: string;
+  let aftrContractInitialState: string;
+  let pstContractSrc: string;
+  let pstInitialState: string;
+
+  let wallet: JWKInterface;
+  let walletAddress: string;
+
+  let arlocal: ArLocal;
+  let warp: Warp;
+  let aftr: Contract<any>;
+  let pst: Contract<any>;
+  let aftrTxId;
+  let aftrSrcTxId;
+  let pstTxId;
+
+  let originalTxId; // set during tests, yeah it sucks, but fuck it for now.
+
+
+  const port = 1966;
+
+  beforeAll(async () => {
+    // note: each tests suit (i.e. file with tests that Jest is running concurrently
+    // with another files has to have ArLocal set to a different port!)
+    arlocal = new ArLocal(port, false);
+    await arlocal.start();
+    LoggerFactory.INST.logLevel('error');
+    //LoggerFactory.INST.logLevel('debug', 'HandlerBasedContract');
+  });
+
+  afterAll(async () => {
+    await arlocal.stop();
+  });
+
+  async function deployContracts() {
+    warp = WarpFactory.forLocal(port);
+    ({jwk: wallet, address: walletAddress} = await warp.testing.generateWallet());
+
+    aftrContractSrc = fs.readFileSync(path.join(__dirname, '../data/aftr/sampleContractSrc.js'), 'utf8');
+    aftrContractInitialState = fs.readFileSync(path.join(__dirname, '../data/aftr/sampleContractInitState.json'), 'utf8');
+
+    pstContractSrc = fs.readFileSync(path.join(__dirname, '../data/aftr/sampleContractSrc.js'), 'utf8');
+    pstInitialState = fs.readFileSync(path.join(__dirname, '../data/aftr/pstInitState.json'), 'utf8');
+
+    ({contractTxId: aftrTxId, srcTxId: aftrSrcTxId} = await warp.createContract.deploy({
+      wallet,
+      initState: aftrContractInitialState,
+      src: aftrContractSrc
+    }));
+
+    ({contractTxId: pstTxId} = await warp.createContract.deployFromSourceTx({
+      wallet,
+      initState: pstInitialState,
+      srcTxId: aftrSrcTxId
+    }));
+
+    aftr = warp
+      .contract<any>(aftrTxId)
+      .setEvaluationOptions({
+        internalWrites: true
+      })
+      .connect(wallet);
+
+    pst = warp
+      .contract<any>(pstTxId)
+      .setEvaluationOptions({
+        internalWrites: true
+      })
+      .connect(wallet);
+
+    await mineBlock(warp);
+  }
+
+  describe('AFTR testcase', () => {
```

Comment:
@arcj0 this is a test written in jest that - I believe - fully recreates a test case from your example.
After deploying contracts, minting tokens, calling "allow" and "deposit", it then creates a fresh Warp instance (https://github.com/warp-contracts/warp/pull/223/files#diff-e85b934a3e7473dbc2261cbc8400e4feef8c1481b9f8c21e914cde8f6c77bd80R252) and reads the token state again.

It also properly verifies the results using "expect".